nautobot/core/testing/filters.py

```diff
@@ -155,6 +170,40 @@ def test_tags_filter(self):
             qs_result = self.queryset.filter(tags=tags[0]).filter(tags=tags[1]).distinct()
             self.assertQuerysetEqualAndNotEmpty(filterset_result, qs_result)
 
+        def test_q_filter_exists(self):
+            """Test the `q` filter exists on a filterset, does not validate the filter works as expected."""
+            self.assertTrue(self.filterset.declared_filters.get("q"))
+
+        def test_q_filter_valid(self):
+            """Test the `q` filter based on attributes in `filter_predicates`."""
+            if not isinstance(self.filterset, SearchFilter):
+                self.skipTest("`q` filter is not a SearchFilter")
+            if not self.filterset.declared_filters.get("q"):
+                self.skipTest("`q` filter not implemented")
+
+            obj = self.queryset.first()
+            for attr, lookup_method in self.get_q_filter().items():
+                if lookup_method not in ["icontains", "iexact"]:
+                    # only testing icontains and iexact filter lookups
+                    continue
+                lookup = str(obj.id)
+                if not attr == "id":
+                    if not hasattr(obj, attr):
+                        # only testing direct attrs, does not support nested lookups
+                        continue
```

Comment:
Feels like this could be a potential bug, if the q filter references something that isn't actually an object attribute. Could we instead *specifically* check for nested lookups via something like `"__" in attr`?

Reply:
Do we want to continue on both conditions or replace with `"__" in attr`?